src/util/symbol_table.h

```diff
@@ -34,6 +34,12 @@ typedef std::multimap<irep_idt, irep_idt> symbol_module_mapt;
                                          it_end=(expr).upper_bound(module); \
       it!=it_end; ++it)
 
+class not_in_symbol_tablet:public std::out_of_range
+{
+public:
+  explicit not_in_symbol_tablet(const irep_idt &id):
+    std::out_of_range("key: " + as_string(id) + " is not in symbol table") {}
+};
```

Comment:
I don't think we need to create yet another out-of-range exception. Unless you can find a good example why we'd be handling just `not_in_symbol_tablet`.

Since we already have a way to "try" to do something, I'd argue for a method that triggers an assertion, should symbol be missing from the symbol table. Something like `std::vector::operator[]` in VC debug mode.

Reply:
The problem with the current way of trying to do something is that it's really verbose.
`symbol_table.lookup(my_key)->get().symbol_type()` is annoying if we know the key exists. My preferred solution would be to have `lookup` return a raw pointer, so that we can just do `symbol_table.lookup(my_key)->symbol_type()`.

Regarding adding a new exception type: it doesn't cost us anything, and it allows us to catch specific out_of_range errors rather than all of them, which seems useful, even if it's not used immediately.